syntax = "proto3";
package golex.virtualtracking;

message RawFrame {
  string schema       = 1;  // "moonalt.raw_v1"
  uint64 frame_id     = 2;
  uint32 width        = 3;
  uint32 height       = 4;
  string pixel_format = 5;  // "BGR24"
  bytes  frame_data   = 6;  // width * height * 3
}

message MaskFrame {
  string schema       = 1;  // "moonalt.mask_v1"
  uint64 frame_id     = 2;
  uint32 width        = 3;
  uint32 height       = 4;
  string pixel_format = 5;  // "GRAY8"
  bytes  frame_data   = 6;  // width * height
}

message Box {
  uint32 x1 = 1;
  uint32 y1 = 2;
  uint32 x2 = 3;
  uint32 y2 = 4;
  float  score = 5;
  int32  class_id = 6; // COCO: person=0, sports ball=32
}

message BoxesFrame {
  string schema   = 1; // "moonalt.boxes_v1"
  uint64 frame_id = 2;
  uint32 width    = 3;
  uint32 height   = 4;
  repeated Box boxes = 5;
}

message BtPacket {
  string schema   = 1; // "moonalt.bt_packet_v1"
  uint64 frame_id = 2;
  RawFrame   raw   = 3;
  MaskFrame  mask  = 4;
  BoxesFrame boxes = 5;
}
